src/adagrams.js

```diff
@@ -1,15 +1,161 @@
+//Function: Introduce letter pool and score chart for use
+const letterPool = {
+  A: 9,
+  B: 2,
+  C: 2,
+  D: 4,
+  E: 12,
+  F: 2,
+  G: 3,
+  H: 2,
+  I: 9,
+  J: 1,
+  K: 1,
+  L: 4,
+  M: 2,
+  N: 6,
+  O: 8,
+  P: 2,
+  Q: 1,
+  R: 6,
+  S: 4,
+  T: 6,
+  U: 4,
+  V: 2,
+  W: 2,
+  X: 1,
+  Y: 2,
+  Z: 1,
+}; 
+
+const scoreChart = {
+  A: 1,
+  B: 3,
+  C: 3,
+  D: 2,
+  E: 1,
+  F: 4,
+  G: 2,
+  H: 4,
+  I: 1,
+  J: 8,
+  K: 5,
+  L: 1,
+  M: 3,
+  N: 1,
+  O: 1,
+  P: 3,
+  Q: 10,
+  R: 1,
+  S: 1,
+  T: 1,
+  U: 1,
+  V: 4,
+  W: 4,
+  X: 8,
+  Y: 4,
+  Z: 10,
+}; 
+
+
 export const drawLetters = () => {
   // Implement this method for wave 1
+  let availableLetters = [];
+  let hand = [];
+
+  for (const [key, value] of Object.entries(letterPool)) {
+    for (let i = 0; i < value; i++) {
+      //add elements with push 
+      availableLetters.push(key); 
+    }
+  }
+
+  for (let i = 0; i < 10; i++) {
+    let letter = availableLetters[Math.floor(Math.random() * availableLetters.length)];
+    hand.push(letter);
+    let index = availableLetters.indexOf(letter); 
+    availableLetters.splice(index, 1);
+  }
+  return hand; 
 };
 
+
 export const usesAvailableLetters = (input, lettersInHand) => {
-  // Implement this method for wave 2
+  // wave 2 
+  // Has two parameters:
+//   `input`, the first parameter, describes some input word, and is a string
+//   - `lettersInHand`, the second parameter, describes an array of drawn letters in a hand. You can expect this to be an array of ten strings, with each string representing a letter
+// - Returns either `true` or `false`
+// - Returns `true` if every letter in the `input` word is available (in the right quantities) in the `lettersInHand`
+// - Returns `false` if not; if there is a letter in `input` that is not present in the `lettersInHand` or has too much of compared to the `lettersInHand`
+
+lettersInHand = [...lettersInHand];
+for (const letter of input) {
+    if(lettersInHand.includes(letter)) {
+      let indexLettersInHand = lettersInHand.indexOf(letter);
+      lettersInHand.splice(indexLettersInHand, 1);
```

Comment:
Notice that we can use the result from `indexOf` itself to tell us whether the value is in the array, since it returns -1 if it is not. So we could wrrite this as
```py
    const indexLettersInHand = lettersInHand.indexOf(letter);
    if (indexLettersInHand !== -1) {
      lettersInHand.splice(indexLettersInHand, 1);
```

Reply:
This approcah (removing the values from from the hand copy as we use them) is going to be a quadratic O(n^2) approach (since the splice, like a remove, must shift everything else forward as things get removes). We could still think about building a frequency table of letters to determine whether the word can be made with the hand in linear time.